guides/advanced-filtering.mdx

```diff
@@ -0,0 +1,245 @@
+  "https://api.app.com/welcome-user", 
+  {
+    filter: {
+      filters: [
+        { path: "event.type", condition: "equals", value: "user_created" },
+        { path: "user.email", condition: "not_equals", value: "test@example.com" },
+        { path: "user.verified", condition: "equals", value: "true" }
+      ],
+      operator: "and"
+    },
+    deduplication: "1h"
+  }
+);
+```
+
+### Marketing Campaigns
+
+Target high-value customers for special offers:
+
+```typescript
+await client.admin.registerPushSubscription(
+  "customer-activity",
+  "vip-offers",
+  "https://marketing.app.com/send-vip-offer",
+  {
+    filter: {
+      filters: [
+        { path: "customer.lifetime_value", condition: "not_equals", value: "0" },
+        { path: "customer.segment", condition: "equals", value: "vip" }
+      ],
+      operator: "or"
+    },
+    rate_limit: "10/d",
+    deduplication: "24h"
+  }
+);
+```
+
+## Rate Limiting
+
+Rate limiting controls how frequently events are delivered to prevent overwhelming your endpoints.
+
+### Format Examples
+
+| Format | Description |
+|--------|-------------|
+| `100/h` | 100 events per hour |
+| `10/m` | 10 events per minute |
+| `5/s` | 5 events per second |
+| `1000/d` | 1000 events per day |
+
+### Use Cases
+
+- **High-volume topics**: Prevent overwhelming downstream services
+- **Cost control**: Limit API calls to external services
+- **Quality of service**: Ensure consistent processing performance
+
+## Deduplication
+
+Deduplication prevents duplicate event deliveries within a specified time window.
+
+### Format Examples
+
+| Format | Description |
+|--------|-------------|
+| `5m` | 5 minutes |
+| `1h` | 1 hour |
+| `24h` | 24 hours |
+| `7d` | 7 days |
+
+### Use Cases
+
+- **Idempotent operations**: Prevent duplicate charges, emails, or notifications
+- **Data consistency**: Avoid processing the same event multiple times
+- **User experience**: Prevent spam notifications
+
+## Schema Configuration
+
+You can also configure advanced filters in your schema files:
+
+```yaml
+subscriptions:
+  - slug: enterprise-alerts
+    topic: system-events
+    type: push
+    endpoint: https://alerts.enterprise.com/webhook
+    filter:
+      filters:
+        - path: severity
+          condition: equals
+          value: critical
+        - path: environment
+          condition: not_equals
+          value: development
+      operator: and
+    rate_limit: "20/m"
+    deduplication: "15m"
+    
+  - slug: user-notifications
+    topic: user-events
+    type: push
+    endpoint: https://notifications.app.com/webhook
+    filter:
+      filters:
+        - path: user.preferences.notifications
+          condition: equals
+          value: enabled
+        - path: event.type
+          condition: equals
+          value: mention
+      operator: and
+    deduplication: "5m"
+```
+
+## Best Practices
+
+### Filter Design
+
+1. **Start simple**: Begin with basic filters and add complexity as needed
```

Comment:
Formatting and rule of 3 feels v AI like - I killed a lot of stuff on the workflows guide because of this!

Reply:
Done